modules/polling/api/fetchPolls.ts

```diff
@@ -162,3 +173,205 @@ export async function getPolls(
+    return { valid: true };
+  }
+
+  const githubPollsHashRes = await fetch(POLLS_HASH_FILE_URL);
+  const githubPollsHashFile = await githubPollsHashRes.json();
+
+  const githubPollsHash: string | undefined = githubPollsHashFile.hash;
+  const cachedPollsHash: string | undefined = await cacheGet(pollsHashCacheKey, network);
+
+  if (githubPollsHash && cachedPollsHash && githubPollsHash === cachedPollsHash) {
+    cacheSet(isPollsHashValidCacheKey, 'true', network, THIRTY_MINUTES_IN_MS);
+    return { valid: true };
+  } else {
+    return { valid: false, hash: githubPollsHash };
+  }
+}
+
+export async function refetchPolls(
+  network: SupportedNetworks,
+  githubHash: string
+): Promise<{ pollList: PollListItem[]; allPolls: Poll[] }> {
+  const allPollsRes = await fetch(AGGREGATED_POLLS_FILE_URL);
+  const allPolls = await allPollsRes.json();
+
+  const pollList = allPolls.map(poll => {
+    const { pollId, startDate, endDate, slug, parameters, summary, title, options, tags } = poll;
+    return {
+      pollId,
+      startDate: new Date(startDate).toISOString(),
+      endDate: new Date(endDate).toISOString(),
+      slug,
+      type: parameters.inputFormat.type,
+      title,
+      summary,
+      options,
+      tags
+    };
+  });
+
+  cacheSet(isPollsHashValidCacheKey, 'true', network, THIRTY_MINUTES_IN_MS);
+  cacheSet(pollsHashCacheKey, githubHash, network, ONE_WEEK_IN_MS);
+  cacheSet(pollListCacheKey, JSON.stringify(pollList), network, ONE_WEEK_IN_MS);
+  allPolls.forEach((poll, i) => {
+    const prevPoll = allPolls[i - 1];
+    const nextPoll = allPolls[i + 1];
+
+    const pollWithLinks = {
+      ...poll,
+      ctx: {
+        prev: prevPoll ? { slug: prevPoll.slug } : null,
+        next: nextPoll ? { slug: nextPoll.slug } : null
+      }
+    };
+
+    const pollCacheKey = getIndividualPollCacheKey(pollWithLinks.pollId);
+    cacheSet(pollCacheKey, JSON.stringify(pollWithLinks), network, ONE_WEEK_IN_MS);
+  });
+
+  return { pollList, allPolls };
+}
+
+export async function getPollList(network: SupportedNetworks): Promise<PollListItem[]> {
+  const { valid: cachedPollsAreValid, hash: githubHash } = await checkCachedPollsValidity(network);
+
+  if (cachedPollsAreValid) {
+    const cachedPollList = await cacheGet(pollListCacheKey, network);
+    if (cachedPollList) {
+      return JSON.parse(cachedPollList);
+    }
+  }
+
+  const { pollList } = await refetchPolls(network, githubHash as string);
+
+  return pollList;
+}
+
+export function filterPollList(
+  pollList: PollListItem[],
+  filters: PollFilterQueryParams
+): Omit<PollsPaginatedResponse, 'tags'> {
+  const {
+    pageSize,
+    page,
+    title: filterTitle,
+    orderBy,
+    tags: filterTags,
+    status,
+    type: filterTypes,
+    startDate: filterStartDate,
+    endDate: filterEndDate
+  } = filters;
+
+  const filteredPolls = pollList
+    // First, filter the poll list by the filters passed
+    .filter(({ title, tags, type, startDate, endDate }) => {
```

Comment:
okay, good code, can you add a unit test to this filter functions ? (i didn't see the changes below this file, so ignore if there's already some unit test)

Reply:
Added!